src/main/java/seedu/ddd/model/contact/common/ContactPredicateBuilder.java

```diff
@@ -70,6 +70,24 @@ public Predicate<Contact> build() throws ParseException {
             Id id = new Id(Integer.parseInt(argMultimap.getValue(PREFIX_ID).get()));
             combinedPredicate = combinedPredicate.and(new ContactIdPredicate(id));
         }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            Phone phoneNumber = new Phone(argMultimap.getValue(PREFIX_PHONE).get());
+            combinedPredicate = combinedPredicate.and(new ContactPhonePredicate(phoneNumber));
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            Email email = new Email(argMultimap.getValue(PREFIX_EMAIL).get());
+            combinedPredicate = combinedPredicate.and(new ContactEmailPredicate(email));
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            String trimmedArgs = argMultimap.getValue(PREFIX_ADDRESS).get().trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.MESSAGE_USAGE));
+            }
```

Comment:
Is it possible to move this to `ListCommandParser` instead for SLAP? If not, its fine too.